packages/babel-jest/src/__tests__/index.ts

```diff
@@ -32,6 +45,85 @@ test(`Returns source string with inline maps when no transformOptions is passed`
   expect(result.map).toBeDefined();
   expect(result.code).toMatch('//# sourceMappingURL');
   expect(result.code).toMatch('customMultiply');
-  expect(result.map.sources).toEqual(['dummy_path.js']);
-  expect(JSON.stringify(result.map.sourcesContent)).toMatch('customMultiply');
+  expect(result.map!.sources).toEqual(['dummy_path.js']);
+  expect(JSON.stringify(result.map!.sourcesContent)).toMatch('customMultiply');
+});
+
+describe('caller option', () => {
+  test('correctly merges from defaults and options', () => {
+    babelJest.process(sourceString, 'dummy_path.js', makeProjectConfig(), {
+      instrument: false,
+      // @ts-ignore
+      supportsDynamicImport: true,
+      supportsStaticESM: true,
+    });
+
+    expect(loadPartialConfig).toHaveBeenCalledTimes(1);
+    expect(loadPartialConfig).toHaveBeenCalledWith(
+      expect.objectContaining({
+        caller: {
+          name: 'babel-jest',
+          supportsDynamicImport: true,
+          supportsStaticESM: true,
+        },
+      }),
+    );
+
+    loadPartialConfig.mockClear();
+
+    babelJest.process(sourceString, 'dummy_path.js', makeProjectConfig(), {
```

Comment:
can we wrap this in another `test` or use `test.each`?

Reply:
yeah, probably